File: backend/src/main/java/com/simone/vehicledealer/domain/Vehicle.java
package com.simone.vehicledealer.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "vehicles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 80)
    private String brand;

    @Column(nullable = false, length = 100)
    private String model;

    @Enumerated(EnumType.STRING)
    @Column(name = "fuel_type", nullable = false, length = 20)
    private FuelType fuelType;

    @Column(nullable = false, length = 50)
    private String color;

    private Integer year;

    @Column(unique = true, length = 17)
    private String chassis;

    @Column(precision = 12, scale = 2)
    private BigDecimal price;

    @Column(name = "external_color", length = 50)
    private String externalColor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dealer_id")
    private Dealer dealer;
}

File: frontend/src/types/vehicle.ts
export const fuelTypes = [
  "GASOLINA",
  "ETANOL",
  "FLEX",
  "DIESEL",
  "ELETRICO",
  "HIBRIDO",
] as const;

export type FuelType = (typeof fuelTypes)[number];

export interface VehicleDealer {
  id: number;
  corporateName: string;
  cnpj: string;
}

export interface Vehicle {
  id: number;
  brand: string;
  model: string;
  fuelType: FuelType;
  color: string;
  year: number | null;
  chassis: string | null;
  price: number | null;
  externalColor: string | null;
  dealer: VehicleDealer | null;
}

export interface VehicleRequest {
  brand: string;
  model: string;
  fuelType: FuelType;
  color: string;
  year: number | null;
  chassis: string | null;
  price: number | null;
  externalColor: string | null;
  dealerId: number | null;
}
